src/OrchardCore/OrchardCore.Sitemaps.Abstractions/Builders/ContentItemsQueryContext.cs

```diff
@@ -1,16 +1,8 @@
-using OrchardCore.ContentManagement;
-
 namespace OrchardCore.Sitemaps.Builders;
 
 public class ContentItemsQueryContext
 {
-    /// <summary>
-    /// Content items to include in sitemap.
-    /// </summary>
-    public IEnumerable<ContentItem> ContentItems { get; set; } = [];
+    public int Skip { get; set; }
```

Comment:
Feels weird to have a class just to hold skip/take arguments.
Could just have added the properties to the methods. Another option would have been to keep this class as-is and just add these two properties.

Reply:
And that would not break the API. 0/0 would mean no paging.

Reply:
It's weird to have a method called GetContentItems that does not return anything. I think the interface make more since now. We could remove the context and replace it with the 2 parameters, but if we ever need to add another property for some reason, then we'll convert back to using context. So keeping the properties in the context seems as a fair approach

Reply:
I personally prefer the option with the object returned, and the skip/take as arguments (last ones)

Reply:
> I personally prefer the option with the object returned, and the skip/take as arguments (last ones)

I second that, and this context object adds an unnecessary allocation in this way too.

As the `ContentItemsQueryContext` is also used by `ISitemapContentItemExtendedMetadataProvider`  as a parameter, I think it would also be fine to leave it as-is. I.e. do not return a `ContentItemsQueryResult` but populate the context as before.

Reply:
@gvkries @sebastienros I think the best option at this time is to undo the breaking change and stick with the existing ugly API.

The only change is that now we have `skip` and `take` parameters that allow us to batch the request using the readonly session.

With this, we should also be able to backport the PR into 1.8